lib/kubeclient/http_proxy_wrapper.rb

```diff
@@ -0,0 +1,49 @@
+module Kubeclient
+  module Common
+    # Wrapper around Net:HTTP to support k8s services
+    class HttpProxyWrapper
+      def initialize(uri, options)
+        @uri = uri
+        @options = options
+      end
+
+      def get(path)
```

Comment:
@jimmidyson I'm not sure I understood what you meant: "should be simple enough to do in one go now I would have thought, delegating to....".